### PR TITLE
Better background detector

### PR DESCRIPTION
**Pull request by egberts**
I made a better background detector based on not only looking at the red element of the RGB, but the total sum of all 3-elements of an RGB, but only for the xterm portion.

Then (for xterm) I performed comparison between foreground RGB and background RGB to see who is "truly" the darker side.

Also, it has some more quirks to deal with gnome-terminal adding escape characters to the end of its ANSI OSC command response to the OSC foreground and OSC background commands.

### CODE REVIEW

File: term-background.bash
Comment (rocky) [inline term-background.bash]: Did you intend to keep the "echo" in there?

File: term-background.bash
Comment (rocky) [inline term-background.bash:178]: since this is bash  and more broadly POSIX shell, I think  `((3711 < VTE_VERSION ))` is the more idomatic, and faster and nicer looking for the part before the `&&`

File: term-background.bash
Comment (egberts) [reply term-background.bash]: Doh!!!!

File: term-background.bash
Comment (egberts) [reply term-background.bash:178]: They are in integer format, we're not comparing strings, I think.

Comment (rocky) [reply term-background.bash:178]: ``((...))`` compares integers as well. Although possibly not POSIX it is I believe in bash, ksh and zsh. See for example: https://www.gnu.org/software/bash/manual/html_node/Shell-Arithmetic.html#index-arithmetic-evaluation

Comment (egberts) [reply term-background.bash:178]: I will defer to you for suggested action.
It is just that JetBrain syntax analyzer pointed this out and suggested a correction. We can go back. It's up to you.

**Review by rocky — CHANGES_REQUESTED** [review]
> I will defer to you for suggested action.

Many thanks. 

> It is just that JetBrain syntax analyzer pointed this out and suggested a correction. We can go back. It's up to you.

Let's go back, thanks. The trouble with standards is that there are so many to choose from. 

I am constantly hounded by this petty stuff. It feels good in at least some little project here that I don't have to put what JetBrains has decided everyone should do.

**Comment by rocky** [comment]
Will change after merge.

**Comment by egberts** [comment]
Are you requesting new pull request with requested changes for multi-shell support, or declining to merge due to lack of test environments?

**Comment by rocky** [comment]
Sure new pull requests are great - bring it on.

This isn't CI tested anywhere yet. CI testing even at a limited scale would be awesome.

**Pull request merged**
